### ISSUE TITLE
RFE: env reset+tenant update and feature availability should be in settings instead of profile

### ISSUE THREAD

**Issue by mishaone** [openshiftio/openshift.io#2307]
I would recommend that now that we have moved out the docs and about links to the ? icon, we should have either a separate cog icon for settings along with those icons or just a separate Settings option under the profile submenu. It feels very unintuitive to me to click on the profile option and then find the environment reset option or where I can choose whether I want production or experimental features, etc.

**Comment by joshuawilson** [comment]
You have a title that does not match the description.
To the point of the title; we have been discussing removing the reset option completely. 
The question of the Settings icon and location is up to UX. -> @catrobson

**Comment by catrobson** [comment]
@mishaone we're moving towards this shortly, we're working on design proposals that we'll be implementing in this train where we will separate profile and settings.

**Comment by mishaone** [comment]
The slightly changed interface makes this a bit higher SEV, IMO. If I didn't know to go into Profile and Edit Profile to find the little Reset Environment option, I would be lost. Particularly since the Settings options have a nice menu of options to do with the account settings but nothing to reset the environment, which would be in frequent use for anyone using OpenShift Online and facing the resource restraints it includes.

**Comment by joshuawilson** [comment]
There are 2 points here. One, it is in the docs for people to find, right?
Two, we don't want/expect users to use the ability often and would like to remove it.

**Comment by mishaone** [comment]
Absolutely, it's in the docs, but docs should be complementary to good UX flow.

Re: removing the option, what is the intended replacement workflow? I use this option after all my OpenShift Online resources are used up or when a project pipeline is consistently failing (separate issue but a fairly common use case in my experience so far, and so not a stretch to consider it may be reasonably common for external users). Thoughts, @joshuawilson?

**Comment by joshuawilson** [comment]
@aslakknutsen @catrobson can you please comment?

**Comment by catrobson** [comment]
@joshuawilson this is an internal only feature IMO.  Is it implemented as such, or is it available to all users?  Once we work out what feature flag level it belongs in, then we can work through flow improvements.

**Comment by mishaone** [comment]
This was a necessary part of the workflow for getting started users, since we guide users through creating two projects. After this, they can't do anything else without reclaiming their OpenShift resources. I can create an issue to track removing this but we need clarification on what the desired workflow is to replace the reset environment option if your OSIO is stuck or you just want to clear the existing projects to work on new ones.

**Comment by catrobson** [comment]
The reason we needed this button was because "Delete space" did not work properly.  As long as "Delete space" works properly, this should be the documented way to clean out your environment to start new projects.

**Comment by joshuawilson** [comment]
I don't think `delete space` fully deletes everything yet.

**Comment by mishaone** [comment]
Thanks, that clears it up a bit. I was looking for an issue tracking this functionality of `delete space` and/or the subsequent removal of `reset environment` but couldn't find anything. Is there something you can point me to so I can monitor it for docs changes, @joshuawilson?

I'll close this issue after having something else to monitor about this issue.

**Comment by mishaone** [comment]
@joshuawilson @catrobson is there any ETA on this feature yet or an issue that I can follow where this issue is being tracked?

We currently still tell users that they can use the reset environment button if they want to reclaim the resources used or if they are permanently stuck and can't fix the issue.

**Comment by joshuawilson** [comment]
@serenamarie125 do you want to move the `reset env`?

**Comment by serenamarie125** [comment]
> @serenamarie125 do you want to move the reset env?

I agree that it should be moved to Settings.  Let me add a design task around this.

**Comment by joshuawilson** [comment]
That makes sense to me. You could also include the Personal Access Token.

**Comment by muruGanesan** [comment]
@serenamarie125 , I will look at this issue and come up with an initial concept tomorrow.

**Comment by muruGanesan** [comment]
@serenamarie125, Here is the first cut UI change.
[I would like to review this proposal with UX review meeting (Monday, Nov 5) today. Based on the input, I will work on the next iteration & share it with Eng]

<img width="1676" alt="image" src="https://user-images.githubusercontent.com/26006817/47976791-4e8a9f00-e0da-11e8-8714-146218889dec.png">





But we need to review a couple of points before move on to next iteration:

1) Current flow:  'Change password' 
It takes the user to "developers .redhat. com" where the user has to re-login again and change the password.  
[UX recommendation: we need to provide a seamless experience]
<img width="1638" alt="image" src="https://user-images.githubusercontent.com/26006817/47976557-06b74800-e0d9-11e8-86e0-f3b1317aaaa8.png">


2) Current flow: 'Reset Environment"
It takes the user to a new page -but the user doesn't have an option 'cancel' or 'come back' to the previous page.
[UX recommendation: User should have an option to 'cancel' the operation and come back to account page]
<img width="1672" alt="image" src="https://user-images.githubusercontent.com/26006817/47976613-3ebe8b00-e0d9-11e8-8ce1-0165d47a3c93.png">

**Comment by sunilmalagi** [comment]
@muruGanesan @serenamarie125 if we go by the alphabetical order, should the 'Account' tab at the beginning of all the remaining tabs?

**Comment by muruGanesan** [comment]
@serenamarie125, thanks Serena!
In fact, created a similar page, later I moved to 'link'. Because we get the repeated text as shown  below:

<img width="758" alt="image" src="https://user-images.githubusercontent.com/26006817/47978621-decde180-e0e4-11e8-9989-ed2b71034d80.png">

**Comment by sunilmalagi** [comment]
1. Account tab re-order 
2. Have the page title called 'Account', like the way when you are in 'Connected Services' tab.
3. I dont think we need have 'Change Password' & 'Reset Environment' as titles,
Only text link OR buttons for 'Change Password' & 'Reset Environment' is enough, if we are going to have buttons, we need know which one is primary and which one is secondary, we also need to decide if we are going to have the description for both.

**Comment by muruGanesan** [comment]
@serenamarie125, 
Here is the Invision link (to review with stakeholders)
https://redhat.invisionapp.com/d/main#/console/15878174/329445923/preview

@ engineering team:
You will NOT have access to the above 'Invision link'. 
I will share the public link, once we complete our internal review.

**Comment by serenamarie125** [comment]
@joshuawilson @christianvogt please take a look at this solution.  We will review at the 12/19 stakeholder meeting as well

**Comment by christianvogt** [comment]
@serenamarie125 @muruGanesan  i cannot open that invision link

**Comment by muruGanesan** [comment]
@joshuawilson, @christianvogt 
Based on our stakeholder's meeting (19Dec18) it will be on hold. 
Points discussed:
1) The users may log-in using GitHub/other social media -change password will take to RedHat developer site
2) Very less % of people change the password.
3) Remove/delete Env- should not be an upfront option for the user
4) Remove/delete Env - role based
5) Remove/delete Env - what happened to the multi-cluster scenario.

@serenamarie125 would discuss with Alexy, Steve Speicher, and other stakeholders get back with further details.

**Comment by muruGanesan** [comment]
Since it is on hold. Let's move to the backlog and removing my assignment
